langserve/playground/src/components/ChatMessageTuplesControlRenderer.tsx

```diff
@@ -0,0 +1,110 @@
+import { withJsonFormsControlProps } from "@jsonforms/react";
+import PlusIcon from "../assets/PlusIcon.svg?react";
+import TrashIcon from "../assets/TrashIcon.svg?react";
+import {
+  rankWith,
+  and,
+  schemaMatches,
+  Paths,
+  isControl,
+} from "@jsonforms/core";
+import { AutosizeTextarea } from "./AutosizeTextarea";
+
+type MessageTuple = [string, string];
+
+export const chatMessagesTupleTester = rankWith(
+  12,
```

Comment:
Not relevant to this PR but might be good to factor these ranking numbers out into a separate place to avoid collisions

Reply:
Could be done, but I think I would've need to pull the rankings manually from the library. Will potentially do later